packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,539 @@
+
+  /**
+   * Retrieves the outputs of the transaction in a format suitable for creating a new transaction.
+   * @returns {Object[]} An array of output objects
+   */
+  public getOutputsForNewTransaction(): { script: Buffer; value: number }[] {
+    return this.originalTx.outs.map((output) => ({
+      script: output.script,
+      value: output.value,
+    }));
+  }
+
+  /**
+   * Estimates the fee required for a successful RBF (Replace-By-Fee) operation.
+   * @returns {string} The estimated RBF fee in satoshis
+   */
+  public estimateRBFFee(): string {
+    const currentFee = this.fee;
+    const vsize = this.vsize;
+
+    const minReplacementFee = BigNumber.max(
+      new BigNumber(currentFee).plus(
+        this.incrementalRelayFee.multipliedBy(vsize),
+      ),
+      new BigNumber(this.rfbFeeRate).multipliedBy(vsize),
+    );
+
+    return String(
+      BigNumber.max(minReplacementFee.minus(currentFee), new BigNumber(1)),
+    );
+  }
+
+  /**
+   * Estimates the fee required for a successful CPFP (Child-Pays-For-Parent) operation.
+   * @returns {string} The estimated CPFP fee in satoshis
+   */
+  public estimateCPFPFee(): string {
+    const originalTxVBytes = this.vsize;
+    const packageSize = originalTxVBytes + ESTIMATED_CHILD_TX_SIZE;
+
+    return new BigNumber(this.cpfpFeeRate).multipliedBy(packageSize).toString();
+  }
+
+  /**
+   * Retrieves the change output of the transaction, if it exists.
+   * @returns {TransactionOutput | null} The change output or null if no change output exists
+   */
+  public getChangeOutput(): TransactionOutput | null {
+    if (this.changeOutputIndex !== undefined) {
+      return this.outputs[this.changeOutputIndex];
+    }
+    return null;
+  }
+
+  // Protected methods
+
+  /**
+   * Analyzes the inputs of the transaction.
+   * @returns {TransactionInput[]} An array of analyzed transaction inputs
+   * @protected
+   */
+  protected analyzeInputs(): TransactionInput[] {
+    return this.originalTx.ins.map((input) => ({
+      txid: input.hash.reverse().toString("hex"),
+      vout: input.index,
+      sequence: input.sequence,
+      scriptSig: input.script.toString("hex"),
+      witness: input.witness.map((w) => w.toString("hex")),
+    }));
+  }
+
+  /**
+   * Analyzes the outputs of the transaction.
+   * @returns {TransactionOutput[]} An array of analyzed transaction outputs
+   * @protected
+   */
+  protected analyzeOutputs(): TransactionOutput[] {
+    return this.originalTx.outs.map((output, index) => ({
+      value: output.value,
+      scriptPubKey: output.script.toString("hex"),
+      address: this.getOutputAddress(output.script),
+      isChange: index === this.changeOutputIndex,
+    }));
+  }
+
+  /**
+   * Attempts to derive the address from an output script.
+   * @param {Buffer} script - The output script
+   * @returns {string} The derived address or an error message if unable to derive
+   * @protected
+   */
+  protected getOutputAddress(script: Buffer): string {
```

Comment:
this could just be a utility function. Doesn't need to be on the class.

Reply:
Added to utils file ...